include/oneapi/dpl/experimental/kt/single_pass_scan.h

```diff
@@ -0,0 +1,413 @@
+            oneapi::dpl::execution::__dpl::make_device_policy(__queue), ::std::forward<_InRange>(__in_rng),
+            ::std::forward<_OutRange>(__out_rng), __n, oneapi::dpl::__internal::__no_op{}, unseq_backend::__no_init_value<_Type>{}, __binary_op, ::std::true_type{});
+    }
+
+    constexpr ::std::size_t __workgroup_size = _KernelParam::workgroup_size;
+    constexpr ::std::size_t __data_per_workitem = _KernelParam::data_per_workitem;
+
+    // Avoid non_uniform n by padding up to a multiple of workgroup_size
+    std::uint32_t __elems_in_tile = __workgroup_size * __data_per_workitem;
+    ::std::size_t __num_wgs = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __elems_in_tile);
+
+    constexpr int __status_flag_padding = SUBGROUP_SIZE;
+    std::uint32_t __status_flags_size = __num_wgs + 1 + __status_flag_padding;
+
+    _FlagStorageType* __status_flags = sycl::malloc_device<_FlagStorageType>(__status_flags_size, __queue);
+    _Type* __status_vals_full = sycl::malloc_device<_Type>(__status_flags_size, __queue);
+    _Type* __status_vals_partial = sycl::malloc_device<_Type>(__status_flags_size, __queue);
+
+    assert(__status_flags && __status_vals_full && __status_vals_partial);
+
+    auto __fill_event = __lookback_init_submitter<_FlagType, _Type, _BinaryOp, _LookbackInitKernel>{}(
+        __queue, __status_flags, __status_vals_partial, __status_flags_size, __status_flag_padding);
+
+    ::std::size_t __current_num_wgs = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __elems_in_tile);
+    ::std::size_t __current_num_items = __current_num_wgs * __workgroup_size;
+
+    auto __prev_event =
+        __lookback_submitter<__data_per_workitem, __workgroup_size, _Type, _FlagType, _LookbackKernel>{}(
+            __queue, __fill_event, __in_rng, __out_rng, __binary_op, __n, __status_flags, __status_flags_size,
+            __status_vals_full, __status_vals_partial, __current_num_items);
+
+    if (0)
+    {
+        return __queue.submit([=](sycl::handler& __hdl) {
+            __hdl.depends_on(__prev_event);
+            __hdl.host_task([=]() { sycl::free(__status_flags, __queue); });
+        });
+    }
+    else
+    {
+        __prev_event.wait();
+        sycl::free(__status_flags, __queue);
+        return __prev_event;
+    }
+}
+
+} // namespace __impl
+
+template <typename _InRng, typename _OutRng, typename _BinaryOp, typename _KernelParam>
+sycl::event
+inclusive_scan(sycl::queue __queue, _InRng&& __in_rng, _OutRng&& __out_rng, _BinaryOp __binary_op,
+               _KernelParam __param = {})
+{
+    return __impl::__single_pass_scan<true>(__queue, ::std::forward<_InRng>(__in_rng), ::std::forward<_OutRng>(__out_rng), __binary_op, __param);
+}
+
+template <typename _InIterator, typename _OutIterator, typename _BinaryOp, typename _KernelParam>
+sycl::event
+inclusive_scan(sycl::queue __queue, _InIterator __in_begin, _InIterator __in_end, _OutIterator __out_begin,
+               _BinaryOp __binary_op, _KernelParam __param = {})
```

Comment:
Do we need an overload with `init` for our current targets?

(If so, this brings up the question of what sort of data do we accept as init:  scalar, device_ptr, single element "range" / iterator?)

Reply:
The plan is to include a version with a single scalar init value to match the functionality of the original scan.

Reply:
OK, so does that need to go in this PR or do you plan to add that in a follow up PR?

Reply:
That's a good question. My goal was to have it implemented in this PR, but it is taking longer to implement + debug correctness than I expected. I am still planning to update this PR with the changes for an initial value, but if it is dragging too long then I might defer it to a follow up PR.